lib/program.js

```diff
@@ -1,16 +1,86 @@
 #!/usr/bin/env node
 import program from 'commander';
+import chalk from 'chalk';
+import Config from 'bcfg';
+const { execSync } = require('child_process');
+const Logger = require('blgr');
+
+import create from './create';
+import install from './install';
+import uninstall from './uninstall';
+import list from './list';
 
-import makePlugin from './makePlugin';
 const { description, version } = require('../package.json');
 
-program
-  .version(version)
-  .description(description);
+const logger = new Logger('info');
+
+const config = new Config('bpanel');
+config.load({
+  env: true,
+  argv: true,
+  arg: true
+});
+
+(async function() {
+  await logger.open();
+
+  program
+    .version(version)
+    .option(
+      '--prefix <path>',
+      'Prefix for module directory defaults to ~/.bpanel'
+    )
+    .description(description);
+
+  program
+    .command('create')
+    .alias('c')
+    .description('Generate a boilerplate bpanel plugin')
+    .action(create(config, logger));
+
+  program
+    .command('list')
+    .alias('l')
+    .description('List all currently installed plugins')
+    .action(list(config, logger));
+
+  program
+    .command('install <name>')
+    .alias('i')
+    .description('Install a plugin for your local bPanel instance.')
+    .option(
+      '-l, --local',
+      'Install as local plugin. If not set, installs from npm.'
+    )
+    .action(install(config, logger));
+
+  program
+    .command('uninstall <name>')
+    .alias('u')
+    .description("Removes all instances of a plugin from your bPanel's config")
+    .option(
+      '-l, --local',
+      'Install as local plugin. If not set, installs from npm.'
+    )
+    .action(uninstall(config, logger));
+
+  program
+    .command('search [keywords]')
+    .alias('s')
+    .description(
+      'Search for available plugins to install'
+    )
+    .action(() => {
+      const output = execSync(`npm search bpanel ${process.argv.slice(3).join(' ')}`, {
```

Comment:
A user could execute arbitrary code here, I don't think its the hugest deal but its still something that we should think about. I could imagine a person giving a malicious command to copy/paste here, it could feel safe because its a `bpanel` command

Reply:
For example, this is no less a risk than if someone gave an uninformed user the command `npm search ; rm -rf ~` to copy and paste into their terminal.